javaslang/src/main/java/javaslang/collection/HashArrayMappedTrie.java

```diff
@@ -101,6 +182,20 @@ static int fromBitmap(int bitmap, int bit) {
 
         abstract AbstractNode<K, V> modify(int shift, int keyHashCode, K key, V value, Action action);
 
+        Iterator<LeafNode<K, V>> nodes() {
+            return new It<>(this);
+        }
+
+        @Override
+        public Iterator<Tuple2<K, V>> iterator() {
+            return nodes().map(node -> Tuple.of(node.key(), node.value()));
```

Comment:
Morning
(please refresh page, PR is updated)

Well, this is good question. If node will be backed by Tuple, we will obtain better performance in iteration along tuples. On other side we'll spend more memory. I'll investigate this idea, but now I would leave this as is